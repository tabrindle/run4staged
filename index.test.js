var childProcess = require('child_process');
var fs = require('fs');
var path = require('path');
var rimraf = require('rimraf');

var run = function run(cmd) {
  return (
    childProcess
      .execSync(cmd, {
        stdio: [0, 'pipe', 'ignore']
      })
      .toString() || ''
  ).trim();
};

describe('run4staged should:', () => {
  beforeEach(() => {
    rimraf.sync(path.resolve(__dirname, 'test'));
    fs.mkdirSync(path.resolve(__dirname, 'test'));
    fs.writeFileSync(path.resolve(__dirname, 'test', 'test.js'), JSON.stringify({ test: 'test' }));
    run('git add test/');
  });

  afterEach(() => {
    run('git reset HEAD ./test || true');
    rimraf.sync(path.resolve(__dirname, 'test'));
  });

  it('run with staged files', () => {
    expect(run('node index.js echo')).toMatchSnapshot();
  });

  it('run with staged files using command arg', () => {
    expect(run('node index.js --command "echo"')).toMatchSnapshot();
  });

  it('run npm script correctly with yarn', () => {
    expect(run('node index.js "yarn npmRunTestScript"')).toContain('YES test/test.js');
  });

  it('run npm script correctly with npm', () => {
    expect(run('node index.js "npm run npmRunTestScript"')).toContain('YES test/test.js');
  });

  it('run with correct staged files given glob', () => {
    fs.writeFileSync(path.resolve(__dirname, 'test', 'test.jsx'));
    run('git add test/');
    expect(run('node index.js echo --glob "**/*.jsx"')).toMatchSnapshot();
  });

  it('run with verbose messaging', () => {
    expect(run('node index.js echo --verbose')).toMatchSnapshot();
  });

  it('display "no command" with verbose', () => {
    expect(run('node index.js --verbose')).toMatchSnapshot();
  });

  it('run each staged file serially', () => {
    fs.writeFileSync(path.resolve(__dirname, 'test', 'test.jsx'));
    run('git add test/');
    expect(run('node index.js echo --serial')).toMatchSnapshot();
  });

  it('run each staged file silently', () => {
    expect(run('node index.js echo --silent')).toMatchSnapshot();
  });
});

describe('run4staged shouldnt:', () => {
  it('run with no staged files', () => {
    rimraf.sync(path.resolve(__dirname, 'test'));
    expect(run('node index.js echo')).toMatchSnapshot();
  });

  it('run with no staged files and warn with verbose', () => {
    rimraf.sync(path.resolve(__dirname, 'test'));
    expect(run('node index.js echo --verbose')).toMatchSnapshot();
  });
})
