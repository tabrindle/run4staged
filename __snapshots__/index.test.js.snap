// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`run4staged should: display "no command" with verbose 1`] = `"No command."`;

exports[`run4staged should: run each staged file serially 1`] = `
"test/test.jsx
test/test.js"
`;

exports[`run4staged should: run each staged file silently 1`] = `"test/test.js"`;

exports[`run4staged should: run with correct staged files given glob 1`] = `"test/test.jsx"`;

exports[`run4staged should: run with staged files 1`] = `"test/test.js"`;

exports[`run4staged should: run with staged files using command arg 1`] = `"test/test.js"`;

exports[`run4staged should: run with verbose messaging 1`] = `
"Staged: test/test.js
Success: echo test/test.js
test/test.js"
`;

exports[`run4staged shouldnt: run with no staged files 1`] = `""`;

exports[`run4staged shouldnt: run with no staged files and warn with verbose 1`] = `"No staged files."`;
